src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private int radius;
+
+    public Circle(String color, int radius) {
+        super(color);
+        this.radius = radius;
+    }
+
+    @Override
+    public double area() {
+        return Math.PI * Math.pow(radius, 2);
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: circle, area: " + area()
+                + " sq. units, radius: " + radius
+                + " units, color: " + getColor());

```

Comment:
The 'draw' method is responsible for too many things, which is a violation of the Single Responsibility Principle. It calculates the area and creates a message to be printed. Consider calculating the area outside of the 'draw' method and passing it as a parameter.